### DISCUSSION TITLE
want to use it with graphql

### DISCUSSION THREAD

**Discussion by mahrozeshahid** [jeremyevans/rodauth#204] [General]
I am starting new project in which I am using Graphql query language for API. So, I am bit confused whether it has good support for Graphql ?

**Comment by Bertg** [comment]
Hi we are using Rodauth (and Rodauth rails) on a Graphql only application.

First, you should not try to do the authentication calls through the graph. Either use the JSON or HTML interface of Rodauth to establish and maintain the sessions. We use the JSON endpoints, as our app is fully written in React.

You can then either use the JWT or pure session approach to handle your GraphQL requests. Let Rodauth do the authentication, and you can then pass the user reference to your Graphql calls.
